frontend/src/components/SubmissionWidget.tsx

```diff
@@ -0,0 +1,144 @@
+// SubmissionWidget.tsx
+import React, { useState } from "react";
+import { SubmissionWidgetProps } from "../models/submission";
+import { Link } from "react-router-dom";
+
+const SubmissionWidget: React.FC<SubmissionWidgetProps> = ({
+  question,
+  globalDocs,
+}) => {
+  const [activeTab, setActiveTab] = useState("submission");
+  const [docsTab, setDocsTab] = useState("question");
+
+  const handleTabSwitch = (tab: "submission" | "docs") => {
+    setActiveTab(tab);
+  };
+
+  const handleDocsTabSwitch = (tab: "question" | "global") => {
+    setDocsTab(tab);
+  };
+
+  const openDocInNewTab = (doc: { content: string; title: string }) => {
+    sessionStorage.setItem("docContent", doc.content);
+    sessionStorage.setItem("docTitle", doc.title);
+    const fullUrl = `${window.location.origin}/markdown-viewer/${doc.title}`;
+    window.open(fullUrl, "_blank");
+  };
+
+  return (
+    <section className="w-full max-w-lg h-[95vh] mx-auto bg-white rounded-lg shadow-md p-4 mt-1 mb-2 lg:ml-auto lg:mr-4 relative">
```

Comment:
Nice edge case. What I think I'll do here is block scrolling on the questions page and only scroll on the markdown section.

Reply:
Resolved, I fixed the bounce effect as well when you overscroll (since we were on the topic of scrolling). Submission widget now has a sticky property that lets it stay on the right side of the screen as the user scrolls down.